scripts/bigquery.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python3
+"""Test the differences in BigQuery schemas between two revisions in schema repository history. This is
+for diagnostic purposes while writing schemas.
+"""
+
+import json
+import os
+import shutil
+import subprocess
+import tempfile
+import argparse
+from contextlib import contextmanager
+from pathlib import Path
+from typing import List, Tuple, Union
+
+import pytest
+
+ROOT = Path(__file__).parent.parent
+
+
+def run(command: Union[str, List[str]], **kwargs) -> str:
+    """Simple wrapper around subprocess.run that returns stdout and raises exceptions on errors."""
+    if isinstance(command, list):
+        args = command
+    elif isinstance(command, str):
+        args = command.split()
+    else:
+        raise RuntimeError(f"run command is invalid: {command}")
+
+    # TODO: log the output
+    return (
+        subprocess.run(args, stdout=subprocess.PIPE, **{**dict(check=True), **kwargs})
+        .stdout.decode()
+        .strip()
+    )
+
+
+def transpile(schema_path: Path) -> dict:
+    """Transpile a JSON Schema into a BigQuery schema."""
+    res = run(
+        [
+            "jsonschema-transpiler",
+            str(schema_path),
+            "--normalize-case",
+            "--resolve",
+            "cast",
+            "--type",
+            "bigquery",
+        ]
+    )
+    schema = json.loads(res)
+    return schema
+
+
+def transform(document: dict) -> dict:
+    """Transform a document for loading into BigQuery."""
+    # TODO: normalize field names using snake casing
+    # TODO: additional properties
+    # TODO: pseudo maps
+    # TODO: anonymous structs
+    raise NotImplementedError()
+
+
+def transpile_schemas(output_path: Path, schema_paths: List[Path]):
+    """Write schemas to directory."""
+    assert output_path.is_dir()
+    for path in schema_paths:
+        namespace, doctype, filename = path.parts[-3:]
+        version = int(filename.split(".")[-3])
+        # pioneer-study schemas were done incorrectly and are ignored here
+        if namespace == "schemas":
+            print(f"skipping {path} due to wrong directory level")
+            continue
+        out = output_path / f"{namespace}.{doctype}.{version}.bq"
+        with out.open("w") as fp:
+            print(f"writing {out}")
+            json.dump(transpile(path), fp, indent=2)
+            fp.write("\n")
+
+
+def load_schemas(input_path: Path):
```

Comment:
```suggestion
def load_schemas(input_path: Path) -> Dict[str, Any]:
```
maybe. 

It looks like this method is currently unused, though it might be nice to keep as a helper function.

Reply:
I've been using the `dict` object directly to annotate, but I'm not sure it's correct in the sense of mypy. I'll keep using it to be consistent.